lib/aws/request.ex

```diff
@@ -324,6 +324,15 @@ defmodule AWS.Request do
       nil ->
         {headers, params}
 
+      param_value when is_map(param_value) ->
```

Comment:
I elected to just use a guard here, but this could be moved into its own function if desired.

Reply:
This is fine by me @jhosteny 👍